File: Project/MyProject/MyProject/Source/MyProject/Private/BasicFrame/Libs/FrameHandle/TickObjectNoPriorityMgr.cpp
#include "MyProject.h"
#include "TickObjectNoPriorityMgr.h"
#include "MClassFactory.h"
#include "MacroDef.h"
#include "Ctx.h"
#include "LogSys.h"
#include "LogTypeId.h"

MY_BEGIN_NAMESPACE(MyNS)

M_IMPLEMENT_AND_REGISTER_CLASS(TickObjectNoPriorityMgr, DelayNoPriorityHandleMgr)

TickObjectNoPriorityMgr::TickObjectNoPriorityMgr()
{

}

void TickObjectNoPriorityMgr::init()
{
	Super::init();
}

void TickObjectNoPriorityMgr::dispose()
{
	Super::dispose();
}

void TickObjectNoPriorityMgr::setClientDispose(bool isDispose)
{

}

bool TickObjectNoPriorityMgr::isClientDispose()
{
	return false;
}

void TickObjectNoPriorityMgr::onTick(float delta, TickMode tickMode)
{
	this->_incDepth();

	this->_onPreAdvance(delta, tickMode);
	this->_onExecAdvance(delta, tickMode);
	this->_onPostAdvance(delta, tickMode);

	this->_decDepth();
}

void TickObjectNoPriorityMgr::_onPreAdvance(float delta, TickMode tickMode)
{

}

void TickObjectNoPriorityMgr::_onExecAdvance(float delta, TickMode tickMode)
{
	int idx = 0;
	int count = this->mNoOrPriorityList->count();
	ITickedObject* tickObject = nullptr;

	while (idx < count)
	{
		tickObject = (ITickedObject*)this->mNoOrPriorityList->get(idx);

		if (nullptr != (IDelayHandleItem*)tickObject)
		{
			if (!((IDelayHandleItem*)tickObject)->isClientDispose())
			{
				tickObject->onTick(delta, tickMode);
			}
		}
		else
		{
			if (MacroDef::ENABLE_LOG)
			{
				GLogSys->log("TickObjectNoPriorityMgr::_onExecAdvance, failed", LogTypeId::eLogCommon);
			}
		}

		++idx;
	}
}

void TickObjectNoPriorityMgr::_onPostAdvance(float delta, TickMode tickMode)
{

}

MY_END_NAMESPACE

File: Project/MyProject/Tool/CSToolKit/Project/ToolKit/ToolKit/Source/Libs/FrameHandle/TickObjectNoPriorityMgr.cs
namespace SDK.Lib
{
    // 每一帧执行的对象管理器
    public class TickObjectNoPriorityMgr : DelayNoPriorityHandleMgr, ITickedObject, INoOrPriorityObject
    {
        public TickObjectNoPriorityMgr()
        {

        }

        override public void init()
        {
            base.init();
        }

        override public void dispose()
        {
            base.dispose();
        }

        public void onTick(float delta, TickMode tickMode)
        {
            this._incDepth();

            this._onPreAdvance(delta, tickMode);
            this._onExecAdvance(delta, tickMode);
            this._onPostAdvance(delta, tickMode);

            this._decDepth();
        }

        virtual protected void _onPreAdvance(float delta, TickMode tickMode)
        {

        }

        virtual protected void _onExecAdvance(float delta, TickMode tickMode)
        {
            int idx = 0;
            int count = this.mNoOrPriorityList.count();
            ITickedObject tickObject = null;

            while (idx < count)
            {
                tickObject = this.mNoOrPriorityList.get(idx) as ITickedObject;

                if (null != (tickObject as IDelayHandleItem))
                {
                    if (!(tickObject as IDelayHandleItem).isClientDispose())
                    {
                        tickObject.onTick(delta, tickMode);
                    }
                }
                else
                {
                    if (MacroDef.ENABLE_LOG)
                    {
                        Ctx.msInstance.mLogSys.log("TickObjectNoPriorityMgr::_onExecAdvance, failed", LogTypeId.eLogCommon);
                    }
                }

                ++idx;
            }
        }

        virtual protected void _onPostAdvance(float delta, TickMode tickMode)
        {

        }
    }
}